Assets/Scripts/Enemies/EnemyAI.cs

```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+using UnityEngine.AI;
+
+public class EnemyAI : MonoBehaviour {
+    [SerializeField] public Transform target;
```

Comment:
Not a big fan of this. I suggest adding an area around the enemy and `OnCollisionEnter` you set the target. And then (optionally) it can be cleared from distance

Reply:
I'll look into it right now

Reply:
I struggled a bit with it but managed to build smt nice